lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -269,6 +270,20 @@ def allocate_load_balancer(action_handler, lb_spec, lb_options, machine_specs)
         end
       end
 
+      # Update load balancer attributes
+      if lb_attributes
+        current = elb.client.describe_load_balancer_attributes(load_balancer_name: actual_elb.name)[:load_balancer_attributes]
+        desired = current.merge(lb_attributes)
```

Comment:
Ruby hash merging is weird - it doesn't do a recursive merge.  So it will depend on what AWS returns:

```ruby
current = {
  :access_log => {
    :enabled => false,
    :s3_bucket_name => "bucket"
  },
  :connection_settings => {
    :idle_timeout => 19
  }
}

lb_attributes = {
  :access_log => {
    :enabled => true
  }
}

# desired would be
desired = {
  :access_log => {
    :enabled => true
  },
  :connection_settings => {
    :idle_timeout => 19
  }
}
```

In this case we lose the bucket name.  I think we should do a recursive merge here.  Check out `deep_merge!` from https://github.com/chef/chef-provisioning-aws/blob/8558aa974d1ac7645c653ee5b6d711500b8dcc3b/lib/chef/provisioning/aws_driver/credentials.rb